### PR TITLE
Refactor to a common bitbake lib

### PR DESCRIPTION
**Pull request by deribaucourt**
Use the new and more complete bitbakeDriver from the client in the language server.

### CODE REVIEW

File: server/src/BitBakeProjectScanner.ts
Comment (WilsonZiweiWang) [inline server/src/BitBakeProjectScanner.ts:1]: Not sure if it is here but the scan results on my side is showing 0 for everything.
all I did was pull the code and relaunch the development host. Also tried `rescan` command.
image

File: server/src/connectionHandlers/onCompletion.ts [hunk 1]
```diff
@@ -54,7 +55,16 @@ export function onCompletionHandler (textDocumentPositionParams: TextDocumentPos
...
+      const bitbakeOverridesCompletionItems: CompletionItem[] = bitBakeProjectScanner.overrides.map(override => {
+        return {
+          label: override,
+          kind: CompletionItemKind.Property,
+          // Present overrides after operators
+          sortText: '~' + override
+        }
+      })
+
+      return bitBakeOperatorCompletionItems.concat(bitbakeOverridesCompletionItems)
```
[the remark this answers is not in the document]
Comment (deribaucourt) [reply server/src/connectionHandlers/onCompletion.ts:72]: Done

File: server/src/BitBakeProjectScanner.ts
Comment (deribaucourt) [reply server/src/BitBakeProjectScanner.ts:1]: Works on my end. it looks like your pathToBuildFolder is incorrectly configured. check your settings.json

Comment (deribaucourt) [reply server/src/BitBakeProjectScanner.ts:1]: There's now a new commit which tests this :)

**Review by WilsonZiweiWang — APPROVED** [review]

**Review by idillon-sfl — APPROVED** [review]

**Pull request merged**